### PR TITLE
Change period and delta using RPC

### PR DESCRIPTION
**Pull request by bubriks**
[#103](https://github.com/Wappsto/wappsto-python/issues/103)

### CODE REVIEW

**Comment by bubriks** [comment]
handlers class could be removed completely

File: wappsto/connection/network_classes/device.py [hunk 1]
```diff
@@ -184,5 +184,5 @@ def delete(self):
 
     def __call_callback(self, event):
         if self.callback is not None:
-            return self.callback(self, event)
+            self.callback(self, event)
         return True
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/device.py]: When does it not return true?

File: wappsto/connection/network_classes/network.py [hunk 2]
```diff
@@ -112,5 +112,5 @@ def __callback_not_set(self, network, event):
 
     def __call_callback(self, event):
         if self.callback is not None:
-            return self.callback(self, event)
+            self.callback(self, event)
         return True
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/network.py]: When does it not return true?

File: wappsto/connection/network_classes/state.py [hunk 3]
```diff
@@ -119,5 +119,5 @@ def delete(self):
 
     def __call_callback(self, event):
         if self.callback is not None:
-            return self.callback(self, event)
+            self.callback(self, event)
         return True
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/state.py]: When does it not return true?

File: wappsto/connection/network_classes/value.py [hunk 4]
```diff
@@ -127,19 +125,20 @@ def set_period(self, period):
...
 
         """
-        try:
-            period = int(period)
-            if period > 0:
-                if self.get_report_state() is not None:
-                    self.period = period
-                    self.__set_timer()
-                    self.wapp_log.debug("Period successfully set.")
+        if period is not None:
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/value.py]: Just return here if period is none

File: wappsto/connection/network_classes/value.py [hunk 5]
```diff
@@ -169,18 +168,19 @@ def set_delta(self, delta):
             delta: Range to report between.
 
         """
-        try:
-            delta = float(delta)
-            if delta > 0:
-                if self.__is_number_type() and self.get_report_state():
-                    self.delta = delta
-                    self.wapp_log.debug("Delta successfully set.")
+        if delta is not None:
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/value.py]: Just return here id delta is None.

File: wappsto/connection/network_classes/value.py [hunk 6]
```diff
@@ -127,19 +125,20 @@ def set_period(self, period):
...
-        try:
-            period = int(period)
-            if period > 0:
-                if self.get_report_state() is not None:
-                    self.period = period
-                    self.__set_timer()
-                    self.wapp_log.debug("Period successfully set.")
+        if period is not None:
+            try:
+                period = int(period)
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/value.py]: The try expect is only for this convertion, and not for the whole block.

File: wappsto/connection/network_classes/value.py [hunk 7]
```diff
@@ -169,18 +168,19 @@ def set_delta(self, delta):
...
         """
-        try:
-            delta = float(delta)
-            if delta > 0:
-                if self.__is_number_type() and self.get_report_state():
-                    self.delta = delta
-                    self.wapp_log.debug("Delta successfully set.")
+        if delta is not None:
+            try:
+                delta = float(delta)
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/value.py]: The try is only for this convertion and not the whole block.

File: wappsto/connection/network_classes/value.py [hunk 8]
```diff
@@ -531,7 +531,7 @@ def delete(self):
 
     def __call_callback(self, event):
         if self.callback is not None:
-            return self.callback(self, event)
+            self.callback(self, event)
         return True
```
Comment (AndreasBomholtz) [inline wappsto/connection/network_classes/value.py]: When does this not return true?

**Pull request merged**